new-lamassu-admin/src/pages/AddMachine/AddMachine.js

```diff
@@ -109,8 +109,16 @@ const initialValues = {
 
 const validationSchema = Yup.object().shape({
   name: Yup.string()
-    .required()
+    .required('Machine name is required.')
     .max(50)
+    .test(
+      'unique-name',
+      'Machine name is already in use.',
+      (value, context) => {
+        if (context.options.context.machineNames.includes(value)) return false
+        return true
+      }
```

Comment:
This should cut a few lines of code

```suggestion
      (value, context) => !context.options.context.machineNames.includes(value)
```